### ISSUE TITLE
No spaces beetween words when parsing pdf

### ISSUE THREAD

**Issue by veepdotai** [smalot/pdfparser#675]
- PHP Version: PHP 8.3.0 (cli) (built: Nov 24 2023 13:48:03) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.3.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.0, Copyright (c), by Zend Technologies

 - PDFParser Version:  "smalot/pdfparser": "^2.8"

### Description:
If I create a document in google docs and download it in pdf format from google, I just get some text witout any spaces when parsing it with pdfparser.

### PDF input

[test-vgwg.pdf](https://github.com/smalot/pdfparser/files/14225204/test-vgwg.pdf)

### Expected output & actual output

input in google docs :
Very good
work guys
Thanks for everything.

download as pdf

I just do the code below

I get the following output:

Verygood
workguys
Thanksforeverything.

### Code

use Smalot\PdfParser\Parser;
$file = "test-vgwg.pdf";
$parser = new Parser(); 
$pdf = $parser->parseFile($file); 
$output = $pdf->getText();
var_dump($output);

**Comment by LaRaye** [comment]
i've been having same issue. it looks like v2.9 does not address this bug. any update?

**Comment by lopatin96** [comment]
Can someone fix it please?

**Comment by GreyWyvern** [comment]
I'm looking at this and with the initial changes that read the OP's doc correctly, I can whittle it down to 3 unit test failures. I'm studying the failures to see if they are actually valid, or if more tweaking is needed.

**Comment by lopatin96** [comment]
@GreyWyvern thank you very much for all your progress invested in the development of this project! I really appreciate it! Please tell me, are there any changes regarding this “bug”?

**Comment by GreyWyvern** [comment]
I'm still working on this. The fix involves using the matrix from `cm` commands as well as the `Td` and `TD` commands. Right now PdfParser only uses them from the `Td` and `TD` commands. However, while just inserting it gets me 98% of the way to a fix, There are two or three unit test PDFs where if I "fix" it for one, the other two break, and vice versa. 😩 

Hopefully soon!

**Comment by LaRaye** [comment]
Got it. Will keep an eye out. Really appreciate all your efforts!!

**Comment by lopatin96** [comment]
@GreyWyvern Got it. Thanks for the info and good luck solving this problem.

**Comment by hgalt** [comment]
@GreyWyvern is there any progress in the meantime?
Due to this problem, the PDFParser is currently useless for me!
I have the feeling, there are not only spaces where are missing, but also \t.
Any workaround before a new version?
Thanks a lot.

**Comment by GreyWyvern** [comment]
Can you try this fork, @hgalt ? https://github.com/GreyWyvern/pdfparser/tree/google-docs

Does it solve your problem? I've boiled down a HUGE amount of changes to the small edits in the fork above. I like the fix (if it solves your issue!) because it's simple, but it has the consequence of adding unnecessary tabs in several other files. Extra tabs is definitely a smaller issue than complete lack of spaces (extra whitespace can easily be stripped by the user), so it might be good to send this as-is as a PR.

**Comment by lopatin96** [comment]
@GreyWyvern For me, additional tabs are not a problem (I can then remove them using my code), it is much worse if there are no spaces between words. Please send it as a PR. And thanks for the work done!

**Comment by LaRaye** [comment]
@GreyWyvern Same here! Difficult to parse text with no whitespace. Extra tabs can be removed. Appreciate your hard work!

**Comment by hgalt** [comment]
@GreyWyvern I do something wrong, because I can not load the branch via composer.
I added under require  `"greywyvern/pdfparser": "google-docs"` and get the error `Could not parse version constraint`

**Comment by hgalt** [comment]
@GreyWyvern got it working without composer.
This fork works for me! Great job, thaks for your effort.

**Comment by gilney-canaltelecom** [comment]
Hi, any news about this?

I'm having the same problem with the following pdf, made using [WPS Office](https://www.wps.com/):

[small_pdf.pdf](https://github.com/user-attachments/files/17019387/small_pdf.pdf)

I tried the google-docs branch and the output still comes without spaces ;/

**Comment by KalmikaV** [comment]
Hello,
(sorry for my poor english)
 just got the same problem. The OCR seems fine if I open it with adobe reader, I can copy and paste the text somewhere just fine, but with pdfparser shows all the words in one line without spaces (you get the enters, line breaks, but not space between words).
 
Is there a solution?

Thanks

**Comment by gilney-canaltelecom** [comment]
@LedSquare No, i guess this project is kinda dead ;/ The clients didn't complain yet, so we are still using this, but we probably will have to change it at some point...

**Comment by LedSquare** [comment]
@gilney-canaltelecom Thanks for answer
